### DISCUSSION TITLE
Does Tailwind work with CSS variables that aren't available at build time?

### DISCUSSION THREAD

**Discussion by tomelliot** [tailwindlabs/tailwindcss#12613] [Help] [answered]
I have a project that builds a static bundle. That bundle is then hosted in an iframe on a 3rd party site. The site provides CSS vars for color theming. I know I can use CSS vars in `tailwind.conf.js`, but I'm not sure if I can use CSS vars that aren't present at build time (only present at run time). Do you know if it that use case will work, or will fail?

My guess is it fails because everything will get purged without the var being defined at build time.

**Comment by wongjn** [comment]
CSS cannot "bleed" into an `<iframe>`. Thus, CSS variables from a 3<sup>rd</sup> party site would not be set inside your project within the `<iframe>`.

**Comment by tomelliot** [comment]
Thanks @wongjn - the 3rd party are taking care of this and injecting the css into the iframe.

**Comment by dario-baumberger** [comment, marked as answer]
Tailwind will generate your css including the not existing css vars. Your site will also apply the styles with the not existing css var. The rules with the not existing vars will be displayed implicit as transparent.

**Reply by tomelliot** [reply]
Thanks @dario-baumberger - good to know

**Reply by dario-baumberger** [reply]
You're welcome :) 

You can also check it in the browser dev tools if you want to prove it.
